persist/fs/read.go

```diff
@@ -380,21 +387,31 @@ func (r *reader) ReadBloomFilter() (*ManagedConcurrentBloomFilter, error) {
 	)
 }
 
-func (r *reader) entryID(id []byte) ident.ID {
-	var idClone checked.Bytes
+func (r *reader) entryClonedBytes(bytes []byte) checked.Bytes {
+	var bytesClone checked.Bytes
 	if r.bytesPool != nil {
-		idClone = r.bytesPool.Get(len(id))
-		idClone.IncRef()
-		defer idClone.DecRef()
+		bytesClone = r.bytesPool.Get(len(bytes))
 	} else {
-		idClone = checked.NewBytes(make([]byte, 0, len(id)), nil)
-		idClone.IncRef()
-		defer idClone.DecRef()
+		bytesClone = checked.NewBytes(make([]byte, 0, len(bytes)), nil)
 	}
+	bytesClone.IncRef()
+	bytesClone.AppendAll(bytes)
+	bytesClone.DecRef()
+	return bytesClone
+}
 
-	idClone.AppendAll(id)
+func (r *reader) entryClonedID(id []byte) ident.ID {
+	return ident.BinaryID(r.entryClonedBytes(id))
+}
 
-	return ident.BinaryID(idClone)
+func (r *reader) entryClonedEncodedTagsTagIter(encodedTags []byte) ident.TagIterator {
```

Comment:
super nit: you can probably make this `entryClonedEncodedTagsIter` so it doesn't stutter as much

Reply:
Sure thing.